indra/sources/phosphoelm/processor.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import requests
+
+from indra.statements import Phosphorylation, Evidence, Agent
+
+gilda_url = 'http://grounding.indra.bio/ground'
+logger = logging.getLogger(__file__)
+
+
+def _gilda_grounder(entity_str):
+    # Try to find a namespace for the enzyme entity string, return the string
+    # that provided the match
+    res = requests.post(gilda_url, json={'text': entity_str})
+    if res.status_code == 200 and res.json():
+        db_ns = res.json()[0]['term']['db']
+        db_id = res.json()[0]['term']['id']
+        return entity_str, db_ns, db_id
+    else:
+        if res.status_code != 200:
+            logger.warning('Gilda service responded with status code %d' %
+                           res.status_code)
+        return entity_str, None, None
+
+
+class PhosphoElmProcessor(object):
+    def __init__(self, file_dump_json=None, keep_empty=False):
+        """The PhosphoElmProcessor processes data dumps from the phospho.ELM
+        database. See http://phospho.elm.eu.org/dataset.html
+
+        file_dump_json : list(dict)
+            JSON compatible list of entries from a phospho.ELM data dump
+        keep_empty : bool
+            If true, also create statements when upstream kinases in
+            entry['kinases'] are not known.
+        """
+        self.statements = []
+        self.statements.extend(self._from_file_dump_json(file_dump_json,
+                                                         keep_empty))
```

Comment:
Most standard processors are implemented such that the constructor doesn't immediately do processing. Rather, the API calls one or more processor functions to populate the statements attribute (e.g. `process_phosphorylations`). The advantage is that the processor can be instantiated without starting any specific processing. I believe this may have to do with my comment on a None input argument to the constructor above.